integration/reloads/helpers.go

```diff
@@ -0,0 +1,189 @@
+package reloads
```

Comment:
Yeah I hear you, I went back and forth on this, it doesn't seem like there's a perfect solution...

To explain a bit, I had this code in the HSM test file originally for handling process reloads and getting the new process after the reload, I pulled it out here to share it with the new test. I wanted to share as much of this code as possible between the new test and the HSM tests so that any flakiness would likely be caught in the new test if it was going to show up in the HSM tests.

I originally wrote this up for the HSM tests instead of using TeleInstance for two reasons:
* TeleInstance does not handle reloads, it holds a main `*service.TeleportProcess` and then another slice of processes for connected nodes. Basically when a reload happens those would all go stale and need to be replaced with the new process. It felt like a bigger effort to add reload support to TeleInstance properly without possibly having unintended effects on other tests.
* TeleInstance overrides the CAs and some identities with static/test ones (that seems to me like the main point of it), while the "real" ones were meant to be under test

I didn't want to put the new test under integration/hsm because:
* it doesn't need to be there, it does nothing with HSMs
* if it gets flaky later because it has caught a production bug, I don't want people to ignore it or get scared off because it's in an HSM package. This way the test is pretty segmented and it's more clear that it's not doing anything really out of the ordinary.

I'm open to any suggestion here, I feel ike the options are:
1. Move the new test under integration/hsm/ and just have the shared code there
2. Move the new test out to integration/ and have it import the shared reload code from integration/hsm
3. Move the new test out to integration/, move the shared code to integration/helpers/reloads/, make new test and HSM tests import from there
4. Hack up TeleInstance to handle process reloads and make both the new test and the HSM tests use that

Reply:
Done in 7608da73ee33c79b50515448f1bd2b09575ac5fc